webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeDataZipHelper.scala

```diff
@@ -31,21 +58,84 @@ trait VolumeDataZipHelper extends WKWDataFormatHelper with ByteUtils {
               }
             }
         }
+    })
+
+  private def withBucketsFromZarr3Zip(zipFile: File)(block: (BucketPosition, Array[Byte]) => Unit): Box[Unit] =
+    for {
+      firstHeaderFilePath <- option2Box(
+        ZipIO.entries(new ZipFile(zipFile)).find(entry => entry.getName.endsWith(Zarr3ArrayHeader.FILENAME_ZARR_JSON)))
+      firstHeaderString <- ZipIO.readAt(new ZipFile(zipFile), firstHeaderFilePath)
+      firstHeader <- JsonHelper.parseAndValidateJson[Zarr3ArrayHeader](firstHeaderString)
+      _ <- firstHeader.assertValid
+      _ <- ZipIO.withUnziped(zipFile) {
+        case (filename, inputStream) =>
+          if (filename.endsWith(Zarr3ArrayHeader.FILENAME_ZARR_JSON)) ()
+          else {
+            parseZarrChunkPath(filename.toString, firstHeader).map { bucketPosition =>
+              val dataCompressed = IOUtils.toByteArray(inputStream)
+              val data = compressor.decompress(dataCompressed)
+              block(bucketPosition, data)
+            }
+          }
+      }
+    } yield ()
+
+  private def parseZarrChunkPath(path: String, zarr3ArrayHeader: Zarr3ArrayHeader): Option[BucketPosition] = {
+    val dimensionNames = zarr3ArrayHeader.dimension_names.getOrElse(Array("x", "y", "z"))
+    val additionalAxesNames: Seq[String] = dimensionNames.toSeq.drop(1).dropRight(3) // drop channel left, and xyz right
```

Comment:
Seems somewhat brittle to me, especially if 2d datasets are also supported. Why not filter the dimensionNames by name like done when exploring?